src/concern/mod.rs

```diff
@@ -39,34 +90,35 @@ pub enum ReadConcern {
     Custom(String),
 }
 
-impl PartialEq for ReadConcern {
-    fn eq(&self, other: &Self) -> bool {
-        self.as_str() == other.as_str()
+impl ReadConcernLevel {
+    pub(crate) fn from_str(s: &str) -> Self {
```

Comment:
this may be useful for users, but I opted to keep it private until we decided that. When exposed publicly, it would probably be better as part of a `From<>` implementation.